Mathlib/AlgebraicGeometry/PrimeSpectrum/Polynomial.lean

```diff
@@ -0,0 +1,221 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
+import Mathlib.LinearAlgebra.Charpoly.BaseChange
+import Mathlib.LinearAlgebra.Eigenspace.Zero
+import Mathlib.RingTheory.AdjoinRoot
+import Mathlib.RingTheory.LocalRing.ResidueField.Ideal
+import Mathlib.RingTheory.TensorProduct.MvPolynomial
+
+/-!
+
+# Prime spectrum of (multivariate) polynomials
+
+Also see `AlgebraicGeometry/AffineSpace` for the affine space over arbitrary schemes.
+
+## Main results
+- `isNilpotent_tensor_residueField_iff`:
+  If `A` is a finite free `R`-algebra, then `f : A` is nilpotent on `κ(𝔭) ⊗ A` for some
+  prime `𝔭 ◃ R` if and only if every non-leading coefficient of `charpoly(f)` is in `𝔭`.
+- `Polynomial.exists_image_comap_of_monic`:
+  If `g : R[X]` is monic, the image of `Z(g) ∩ D(f) : Spec R[X]` in `Spec R` is compact open.
+- `Polynomial.isOpenMap_comap_C`: The structure map `Spec R[X] → Spec R` is an open map.
+- `MvPolynomial.isOpenMap_comap_C`: The structure map `Spec R[X] → Spec R` is an open map.
```

Comment:
Maybe `R[X]` isn't the right notation in this setting.
I actually didn't know that this result was true without finiteness conditions on the number of variables. In particular, this doesn't follow from the standard lemmas (flat + fp => open, etc)

Reply:
I changed it to `R[X̲]`. Hopefully it's better.

Yeah I didn't know this was true as well until Lean didn't ask for a finiteness assumption.